<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DocumentWindow</class>
 <widget class="QMainWindow" name="DocumentWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1804</width>
    <height>1104</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Telemetry Viewer</string>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="m_root_widget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="m_splitter">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="content_container" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="topbar" native="true">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>32</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>32</height>
           </size>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>11</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>11</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Installation:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="m_installation_selector"/>
           </item>
           <item>
            <widget class="QComboBox" name="m_memory_scaling">
             <item>
              <property name="text">
               <string>Bytes</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Kilobytes</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Megabytes</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Gigabytes</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="m_mode_selector">
             <item>
              <property name="text">
               <string>Timeline (Raw)</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Timeline (Running average)</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Distribution</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page_0">
           <attribute name="title">
            <string>Samples</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="ChartWidget" name="m_chart_view"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="m_timeline_page">
           <attribute name="title">
            <string>Events</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QSplitter" name="splitter_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="orientation">
               <enum>Qt::Orientation::Vertical</enum>
              </property>
              <widget class="TimelineWidget" name="m_timeline_widget">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>1</verstretch>
                </sizepolicy>
               </property>
              </widget>
              <widget class="QTreeWidget" name="m_timeline_tree">
               <column>
                <property name="text">
                 <string>Event</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Duration</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Path</string>
                </property>
               </column>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="bottombar" native="true">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>32</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>32</height>
           </size>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="leftMargin">
            <number>11</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>11</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Range:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="m_event_picker">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>240</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Policy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="end_label">
             <property name="text">
              <string>Start:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="TimePickerWidget" name="m_start_edit" native="true">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Policy::Fixed</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="start_label">
             <property name="text">
              <string>End:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="TimePickerWidget" name="m_end_edit" native="true">
             <property name="minimumSize">
              <size>
               <width>80</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QSplitter" name="m_splitter_vertical">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>480</width>
         <height>0</height>
        </size>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <property name="childrenCollapsible">
        <bool>false</bool>
       </property>
       <widget class="QTreeWidget" name="m_overview_view">
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::NoSelection</enum>
        </property>
        <attribute name="headerDefaultSectionSize">
         <number>200</number>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">Statistic</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Value</string>
         </property>
        </column>
       </widget>
       <widget class="QTreeWidget" name="m_providers_view">
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::NoSelection</enum>
        </property>
        <attribute name="headerDefaultSectionSize">
         <number>200</number>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">Provider</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Name</string>
         </property>
        </column>
       </widget>
       <widget class="QChartView" name="m_statistics_view">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>300</height>
         </size>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1804</width>
     <height>33</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="m_menu_recents">
     <property name="title">
      <string>Recent Traces...</string>
     </property>
     <addaction name="m_action_clear_recents"/>
    </widget>
    <addaction name="m_action_new"/>
    <addaction name="m_action_open"/>
    <addaction name="m_menu_recents"/>
    <addaction name="separator"/>
    <addaction name="m_action_save"/>
    <addaction name="separator"/>
    <addaction name="m_action_close"/>
    <addaction name="separator"/>
    <addaction name="m_action_exit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="m_action_about"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="m_run_fps_test"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::AboutRole</enum>
   </property>
  </action>
  <action name="m_action_open">
   <property name="text">
    <string>Open Trace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="m_action_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="m_action_exit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="m_action_save">
   <property name="text">
    <string>Save Trace</string>
   </property>
  </action>
  <action name="m_action_new">
   <property name="text">
    <string>New Window</string>
   </property>
  </action>
  <action name="m_action_clear_recents">
   <property name="text">
    <string>Clear Menu</string>
   </property>
  </action>
  <action name="m_action_close">
   <property name="text">
    <string>Close Trace</string>
   </property>
  </action>
  <action name="m_run_fps_test">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::MediaPlaybackStart"/>
   </property>
   <property name="text">
    <string>Run FPS Test</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TimePickerWidget</class>
   <extends>QWidget</extends>
   <header>widgets/TimePickerWidget.h</header>
   <container>1</container>
   <slots>
    <signal>value_changed(int)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>TimelineWidget</class>
   <extends>QGraphicsView</extends>
   <header>widgets/TimelineWidget.h</header>
   <slots>
    <slot>zoomIn()</slot>
    <slot>zoomOut()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">QtCharts/QtCharts</header>
  </customwidget>
  <customwidget>
   <class>ChartWidget</class>
   <extends>QGraphicsView</extends>
   <header>widgets/ChartWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_run_fps_test</sender>
   <signal>triggered()</signal>
   <receiver>DocumentWindow</receiver>
   <slot>run_fps_test()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_action_open</sender>
   <signal>triggered()</signal>
   <receiver>DocumentWindow</receiver>
   <slot>open_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_action_save</sender>
   <signal>triggered()</signal>
   <receiver>DocumentWindow</receiver>
   <slot>save_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_start_edit</sender>
   <signal>value_changed(int)</signal>
   <receiver>DocumentWindow</receiver>
   <slot>range_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1110</x>
     <y>1059</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_end_edit</sender>
   <signal>value_changed(int)</signal>
   <receiver>DocumentWindow</receiver>
   <slot>range_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1265</x>
     <y>1059</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_event_picker</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>DocumentWindow</receiver>
   <slot>event_range_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>1060</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_action_close</sender>
   <signal>triggered()</signal>
   <receiver>DocumentWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_providers_view</sender>
   <signal>itemChanged(QTreeWidgetItem*,int)</signal>
   <receiver>DocumentWindow</receiver>
   <slot>provider_item_changed(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1563</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_action_new</sender>
   <signal>triggered()</signal>
   <receiver>DocumentWindow</receiver>
   <slot>new_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_action_clear_recents</sender>
   <signal>triggered()</signal>
   <receiver>DocumentWindow</receiver>
   <slot>clear_recent_items()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>901</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>run_fps_test()</slot>
  <slot>open_file()</slot>
  <slot>save_file()</slot>
  <slot>range_changed()</slot>
  <slot>event_range_changed(int)</slot>
  <slot>populate_recent_items()</slot>
  <slot>provider_item_changed(QTreeWidgetItem*)</slot>
  <slot>clear_recent_items()</slot>
  <slot>new_file()</slot>
 </slots>
</ui>
